### ISSUE TITLE
`binde` with mouse repeats even if the button isn't held down

### ISSUE THREAD

**Issue by cd-CreepArghhh** [hyprwm/Hyprland#5671]
### Hyprland Version

<details> <summary>System/Version info</summary>

TL;DR: Arch Linux using `hyprland` (not `-git`), updated 5 minutes ago, hardened kernel, no plugins or flags.

```sh
Hyprland, built from branch  at commit fe7b748eb668136dd0558b7c8279bfcd7ab4d759  (props: bump version to 0.39.1).
Date: Tue Apr 16 16:01:03 2024
Tag: v0.39.1, commits: 4460

flags: (if any)


System Information:
System name: Linux
Node name: ArchLinux
Release: 6.8.7-hardened1-2-hardened
Version: #1 SMP PREEMPT_DYNAMIC Wed, 17 Apr 2024 22:21:16 +0000


GPU information: 
03:00.0 VGA compatible controller [0300]: Advanced Micro Devices, Inc. [AMD/ATI] Barcelo [1002:15e7] (rev c5) (prog-if 00 [VGA controller])


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:
```

</details>

### Bug or Regression?

Bug

### Description

I learn about `binde` and I immediately add it to the volume keybindings I have. For the volume **buttons**, it works perfectly. However, I also use mouse buttons 4 and 5 to change the volume, and pressing it once results in the volume increasing continuously. Pressing the other button makes it decrease continuously. It won't stop until I release the modifier key, or if there's none set pressing another key stops it.

A couple minutes later I try it with mouse scrolling, and get the same result.

See the videos attached below for more info.

### How to reproduce

Here's my volume keybindings in case you have extra mouse buttons that you can try it with:
```
# Make mouse buttons 4 and 5 change volume
bind = , mouse:275, exec, pamixer -d 2
bind = , mouse:276, exec, pamixer -i 2
```

And here's the one with mouse scrolling (**EPILEPSY WARNING**)
```
# Scroll through existing workspaces with mainMod + scroll
bind = $mainMod, mouse_down, workspace, e+1
bind = $mainMod, mouse_up, workspace, e-1
```

#### Volume:
![Video](https://github.com/hyprwm/Hyprland/assets/65649991/185e91cb-3f6b-435b-93dc-769a0f13b323)
Look at the yellow thing in my bar to see it change.
In the video, I:
  - Press mouse 5 and mouse 4 a couple of times.
  - Edit the config file.
  - Press mouse 5 *once*. Then mouse 4 *once*. I repeat this once.

#### Window scrolling (<ins>EPILEPSY WARNING</ins>):
![Video](https://github.com/hyprwm/Hyprland/assets/65649991/940637a0-0d00-4fed-abb0-0685d8e073df)
Sorry about the lag, my OBS settings are a bit funky.
In the video, I:
  - Hold SUPER and scroll up/down a couple of times.
  - Edit the config file.
  - Hold SUPER and scroll up/down *one notch*.
  - Wait a bit, then release SUPER. Repeat a couple of times.
  - Edit the config file again.
  - Hold SUPER and scroll up/down a couple of times again.

#### EDIT: found an... uh, *interesting* use case...
```
binden = SUPER CTRL_SHIFT, mouse:272, exec, ydotool click 00 40 80
```
![Video](https://github.com/hyprwm/Hyprland/assets/65649991/0087feb9-7d5f-4ef0-8100-51867d1069fd)
Again, I'm only clicking once, and I'm not holding down the mouse button.

**Comment by LowEntropyEntity** [comment]
I am seeing the same or very similar behavior in 0.47.2.

### Hyprland version

<details> <summary>System/Version info</summary>

```
Hyprland 0.47.2 built from branch  at commit 882f7ad7d2bbfc7440d0ccaef93b1cdd78e8e3ff  (version: bump to 0.47.2).
Date: Sun Feb 2 00:47:17 2025
Tag: v0.47.2, commits: 5767
built against:
 aquamarine 0.7.2
 hyprlang 0.6.0
 hyprutils 0.5.0
 hyprcursor 0.1.11
 hyprgraphics 0.1.2


no flags were set


System Information:
System name: Linux
Node name: enki
Release: 6.13.2-arch1-1
Version: #1 SMP PREEMPT_DYNAMIC Sat, 08 Feb 2025 18:54:55 +0000


GPU information:
00:02.0 VGA compatible controller [0300]: Intel Corporation Alder Lake-HX GT1 [UHD Graphics 770] [8086:4688] (rev 0c) (prog-if 00 [VGA controller])
01:00.0 VGA compatible controller [0300]: NVIDIA Corporation AD104GLM [RTX 3500 Ada Generation Laptop GPU] [10de:27bb] (rev a1) (prog-if 00 [VGA controller])
NVRM version: NVIDIA UNIX x86_64 Kernel Module  570.86.16  Fri Jan 24 21:25:51 UTC 2025


os-release: NAME="Arch Linux"
PRETTY_NAME="Arch Linux"
ID=arch
BUILD_ID=rolling
ANSI_COLOR="38;2;23;147;209"
HOME_URL="https://archlinux.org/"
DOCUMENTATION_URL="https://wiki.archlinux.org/"
SUPPORT_URL="https://bbs.archlinux.org/"
BUG_REPORT_URL="https://gitlab.archlinux.org/groups/archlinux/-/issues"
PRIVACY_POLICY_URL="https://terms.archlinux.org/docs/privacy-policy/"
LOGO=archlinux-logo


plugins:


```

</details>

keybinds:
```
binde = , mouse:275, sendshortcut, , down,
binde = , mouse:276, sendshortcut, , up,
```

### How to reproduce

1. open an app that supports scrolling, e.g. chromium browser on this page
2. click mouse's back button to scroll up (or hold it to continuously scroll up)
3. release mouse button
   * content stops scrolling
5. move mouse/cursor up, still with no buttons pressed
   * **expected**: content remains static
   * **actual**:      content resumes scrolling

The mouse's forward behavior exhibits the same behavior, scrolling down even after released

**Comment by 00-kat** [comment]
I personally can't reproduce this anymore (on Hyprland 0.47.2[^1]).

[^1]: I'm not including the detailed `--version` output as it's the same as you except `hyprgraphics` which is at 0.1.1 for me.
